common/vm_wait_guest_hostname.yml

```diff
@@ -29,13 +34,30 @@
   retries: "{{ ((vm_get_hostname_timeout | default(300) | int) / 5) | int }}"
   register: vm_guest_facts
   until:
-    - "'instance' in vm_guest_facts"
-    - vm_guest_facts.instance.guest.toolsRunningStatus == "guestToolsRunning"
-    - vm_guest_facts.instance.guest.hostName
+    - vm_guest_facts.instance.guest.toolsRunningStatus | default('') == "guestToolsRunning"
+    - vm_guest_facts.instance.guest.hostName | default('')
     - (not wait_guest_hostname) or (wait_guest_hostname and wait_guest_hostname == vm_guest_facts.instance.guest.hostName)
+  ignore_errors: true
 
-- name: Display the retrieved guest info
+- name: "Display the retrieved guest info"
   ansible.builtin.debug: var=vm_guest_facts
-  when: enable_debug | bool
-- ansible.builtin.debug:
+  when: enable_debug | default(false)
```

Comment:
You can set 'enable_debug' parameter value in the "vars:" of your playbook if want to test task only.